interpreter/text/arrange.ml

```diff
@@ -444,9 +444,9 @@ let nan = function
 
 let result res =
   match res.it with
-  | ValueResult (LitResult lit) -> literal lit
-  | SimdF32x4Result _ -> failwith "unimplemented"
-  | ValueResult (NanResult nanop) ->
+  | NumResult { it = LitPat lit; _ } -> literal lit
+  | SimdResult _ -> failwith "unimplemented"
+  | NumResult { it = NanPat nanop; _ } ->
```

Comment:
Here I would actually create a separate function for printing num_pat's.

Reply:
Done.